crux-lucene/src/crux/lucene.clj

```diff
@@ -262,15 +262,18 @@
                             (txc/conformed-tx-events->doc-hashes conformed-tx-events))]
     (reduce (fn [acc {:keys [op] :as tx-event}]
               (case op
-                :crux.tx/evict (-> acc (update :evicted-eids conj (:eid tx-event)))
+                :crux.tx/evict (let [evicted-eids (conj (:evicted-eids acc) (:eid tx-event))]
+                                 (-> (apply dissoc acc :docs evicted-eids)
```

Comment:
I don't think this does do what you want it to - try adding puts of other entities in the transaction and check whether those docs still end up in Lucene? i.e. put a, evict a, put b, query for b.

Reply:
I just extended the test to try to capture your suggestion (which passes) - does it help?